compiler/rustc_ast/src/expand/allocator.rs

```diff
@@ -1,18 +1,17 @@
 use rustc_span::symbol::{sym, Symbol};
 
-#[derive(Clone, Debug, Copy, HashStable_Generic)]
+#[derive(Clone, Debug, Copy, Eq, PartialEq, HashStable_Generic)]
 pub enum AllocatorKind {
     Global,
     Default,
 }
 
-impl AllocatorKind {
-    pub fn fn_name(&self, base: Symbol) -> String {
-        match *self {
-            AllocatorKind::Global => format!("__rg_{}", base),
-            AllocatorKind::Default => format!("__rdl_{}", base),
-        }
-    }
+pub fn global_fn_name(base: Symbol) -> String {
+    format!("__rust_{}", base)
```

Comment:
Can you also update the comments elsewhere in the compiler+stdlib code to reflect the new structure being deployed here?

Here are the references my naive grep yielded:

https://github.com/rust-lang/rust/blob/99da9aec246e6b500a4e589e4b9096306bb89b07/library/alloc/src/alloc.rs#L23-L27

https://github.com/rust-lang/rust/blob/99da9aec246e6b500a4e589e4b9096306bb89b07/library/alloc/src/alloc.rs#L354-L356

https://github.com/rust-lang/rust/blob/297a8018b525c28ef10ee6a91d61954839b508b9/compiler/rustc_monomorphize/src/partitioning/default.rs#L517-L521